### ISSUE TITLE
TZ issue

### ISSUE THREAD

**Issue by alsam** [mfreeborn/heliocron#41]
hello,
i'm not able to figure out how to set different time zones. for me it just doesn't work as expected:

```sh
cargo run -- --latitude 55.7558N --longitude 37.6173E  report # Moscow
LOCATION
--------
Latitude: 55.7558N
Longitude: 37.6173E

DATE
----
2022-01-18 12:00:00 +03:00

Solar noon is at:         2022-01-18 12:39:53 +03:00
The day length is:        7h 48m 15s

Sunrise is at:            2022-01-18 08:45:46 +03:00
Sunset is at:             2022-01-18 16:34:01 +03:00
...
```
so good so far,

```sh
TZ=EST cargo run -- --latitude 40.7128N --longitude 74.0060W  report # New York
LOCATION
--------
Latitude: 40.7128N
Longitude: 74.0060W

DATE
----
2022-01-18 12:00:00 -05:00

Solar noon is at:         2022-01-18 12:06:29 -05:00
The day length is:        9h 40m 11s

Sunrise is at:            2022-01-18 07:16:24 -05:00
Sunset is at:             2022-01-18 16:56:35 -05:00
...
```
ok

```sh
cargo run -- --latitude 40.7128N --longitude 74.0060W  -t -05:00 report # New York
LOCATION
--------
Latitude: 40.7128N
Longitude: 74.0060W

DATE
----
2022-01-18 12:00:00 +03:00

Solar noon is at:         2022-01-18 20:06:23 +03:00
The day length is:        9h 39m 38s

Sunrise is at:            2022-01-18 15:16:34 +03:00
Sunset is at:             2022-01-19 00:56:12 +03:00
...
```
**doesn't work** `-t -05:00` option

```sh
TZ=PST cargo run -- --latitude 37.7749N --longitude 122.4194W  report # San Francisco
LOCATION
--------
Latitude: 37.7749N
Longitude: 122.4194W

DATE
----
2022-01-18 12:00:00 +00:00

Solar noon is at:         2022-01-18 20:20:05 +00:00
The day length is:        9h 54m 43s

Sunrise is at:            2022-01-18 15:22:43 +00:00
Sunset is at:             2022-01-19 01:17:26 +00:00
...
```
**doesn't work** even with `TZ` set to seemed correct `PST` time zone

could you please point me to where what i missed something? Thanks!

**Comment by mfreeborn** [comment]
I can't explain the behaviour with some of the `TZ` values working, whilst others do not... I don't explicitly depend on `TZ` anywhere in the code... It's also giving some pretty incorrent calculations. I'll need to look into that later.

I have, however, identified the bug which means the `-t` flag is not being honoured: the program is only parsing the `-t` flag if `-d` is _also_ set on the command line. See here: https://github.com/mfreeborn/heliocron/blob/5eb266f3632932a929b5b9f8babc0963c3b6c807/src/config.rs#L176-L182

**Comment by mfreeborn** [comment]
I've now fixed this in #50, which will be part of a new release likely later today.

Specifically, code block #3 in your original post is now fixed.

As for code block #4, that's to do with valid values for the TZ environment variable. If you use `tzselect` and work your way through the options for west coast USA, it'll tell you the time zone `America/Los_Angeles`.

Replacing `TZ=PST` with `TZ=America/Los_Angeles` gives  you the behaviour you are expecting in code block #4.
